### PR TITLE
perf(spec-parser): only allow single parameter to manifest for post

### PR DESCRIPTION
**Pull request by SLdragon**
1. Each post body property should be a parameter
2. Optional parameter should not map to manifest
3. Resolve server url environment variables

### CODE REVIEW

File: packages/fx-core/src/common/spec-parser/utils.ts [hunk 1]
```diff
@@ -78,20 +77,19 @@ export function isSupportedSchema(schema: OpenAPIV3.SchemaObject): boolean {
...
+      const result = checkPostBodyRequiredParameters(
+        properties[property] as OpenAPIV3.SchemaObject
+      );
+      count += result;
     }
-    return true;
+    return count;
   } else {
-    return false;
+    return NaN;
```
Comment (yuqizhou77) [inline packages/fx-core/src/common/spec-parser/utils.ts:92]: Is this for the case type:array?

Comment (SLdragon) [reply packages/fx-core/src/common/spec-parser/utils.ts:92]: yes, schema.array, schema.allOf, schema.oneOf, schema.not, and schema.anyOf

File: packages/fx-core/src/common/spec-parser/utils.ts [hunk 2]
```diff
@@ -78,20 +77,19 @@ export function isSupportedSchema(schema: OpenAPIV3.SchemaObject): boolean {
...
+    let count = 0;
     for (const property in properties) {
-      const result = isSupportedSchema(properties[property] as OpenAPIV3.SchemaObject);
-      if (!result) {
-        return false;
-      }
+      const result = checkPostBodyRequiredParameters(
+        properties[property] as OpenAPIV3.SchemaObject
+      );
+      count += result;
```
Comment (yuqizhou77) [inline packages/fx-core/src/common/spec-parser/utils.ts:88]: thought there might cause exception? the value can be NaN returned?

Comment (SLdragon) [reply packages/fx-core/src/common/spec-parser/utils.ts:88]: NaN + any number would still NaN, so it would work

Comment (yuqizhou77) [reply packages/fx-core/src/common/spec-parser/utils.ts:88]: do we validate such cases in validate() today?

Comment (yuqizhou77) [reply packages/fx-core/src/common/spec-parser/utils.ts:88]: oh, realized we will filter out these APIs.

**Review by yuqizhou77 — DISMISSED** [review]

**Review by adashen — APPROVED** [review]

**Pull request merged**